src/hooks/useOrders.ts

```diff
@@ -39,39 +35,47 @@ export function useOrders(): Result {
   //   2. remove any orders that may have been re-sent at a higher gas price
-  const currentPendingOrders: PendingTxObj[] =
-    (networkId && userAddress && pendingOrders[networkId] && pendingOrders[networkId][userAddress]) || emptyArray
+  const currentPendingOrders = (userAddress && networkId && pendingOrders[networkId]?.[userAddress]) || EMPTY_ARRAY
   useEffect(() => {
     // Don't fire if there are no pending orders...
-    if (networkId && userAddress && currentPendingOrders.length > 0) {
+    if (userAddress && networkId && currentPendingOrders.length > 0) {
       const managePendingOrders = async (): Promise<void> => {
-        const [latestBlock, pendingOrderStatuses] = await Promise.all([
-          web3.eth.getBlock(blockNumber || 'latest'),
-          Promise.all(currentPendingOrders.map(({ txHash }) => web3.eth.getTransaction(txHash))),
-        ])
-
-        const transactionsToRemoveArray = pendingOrderStatuses
-          // order.status === null? transaction has been dropped
-          // add latter txHash to arr of removable transactions
-          .reduce<string[]>(
-            (acc, status, index) => (!status ? acc.concat(currentPendingOrders[index].txHash) : acc),
-            latestBlock.transactions,
-          )
+        const latestBlock = await web3.eth.getBlock(blockNumber || 'latest')
 
-        const transactionsSet = new Set(transactionsToRemoveArray)
+        // Set from block transaction array for easy lookup
+        const transactionsSet = new Set(latestBlock.transactions)
 
-        const filteredPendingOrders = currentPendingOrders.filter(
+        // check CURRENT pending orders txHash against the new block's
+        // mined transaction list - return orders still pending
+        let blockTransactionsFilteredPendingOrders = currentPendingOrders.filter(
           ({ txHash }: { txHash: string }) => !transactionsSet.has(txHash),
         )
 
+        // if some orders are still pending
+        if (blockTransactionsFilteredPendingOrders.length !== 0) {
+          // can return Transaction receipt as Pending/Mined OR null
+          // Pending TransactionReceipt will not have blockNumber/Hash or transactionIndex
+          // Mined TransactionRceipt WILL have the above props filled
+          // null indicates dropped tx
+          const transactionStatusArray = await Promise.all(
+            blockTransactionsFilteredPendingOrders.map(order => web3.eth.getTransaction(order.txHash)),
+          )
+
+          // Remove orders that are NULL and/or orders that have a blockNumber (indicating mined status)
+          // e.g [null, { blockNumber: 19082137, ... }, ...]
+          blockTransactionsFilteredPendingOrders = blockTransactionsFilteredPendingOrders.filter(
+            (_, index) => !!transactionStatusArray[index] && !transactionStatusArray[index].blockNumber,
+          )
+        }
+
         // don't dispatch unneccessarily
```

Comment:
Maybe say: Dispatch if a pending transaction was mined instead of "don't dispatch unneccessarily"